project-clone/internal/git_operations.go

```diff
@@ -35,9 +35,21 @@ func CloneProject(project *dw.Project) (*git.Repository, error) {
 	}
 
 	var defaultRemoteName, defaultRemoteURL string
-
 	if project.Git.CheckoutFrom != nil {
-		defaultRemoteName = project.Git.CheckoutFrom.Remote
+		// we can omit remote attribute only if there is a single remote
+		if len(project.Git.CheckoutFrom.Remote) == 0 {
+			if len(project.Git.Remotes) == 1 {
+				// take the first one
+				for remoteName, remoteUrl := range project.Git.Remotes {
+					defaultRemoteName, defaultRemoteURL = remoteName, remoteUrl
+				}
+			} else {
+				// need to specify
+				return nil, fmt.Errorf("project checkoutFrom remote can only omitted if there is only one remote")
+			}
+		} else {
+			defaultRemoteName = project.Git.CheckoutFrom.Remote
+		}
```

Comment:
Maybe a little cleaner
```suggestion
		defaultRemoteName := project.Git.CheckoutFrom.Remote
		if defaultRemoteName == "" {
			if len(project.Git.Remotes) == 1 {
				for remoteName := range project.Git.Remotes {
					defaultRemoteName = remoteName
				}
			} else {
				// need to specify
				return nil, fmt.Errorf("project checkoutFrom remote can only omitted if there is only one remote")
			}
		}
```